desktop-widgets/mainwindow.cpp

```diff
@@ -612,7 +604,7 @@ void MainWindow::on_actionCloudOnline_triggered()
 	git_local_only = isOffline;
 	if (!isOffline) {
 		// User requests to go online. Try to sync cloud storage
-		if (unsavedChanges()) {
+		if (!Command::isClean()) {
```

Comment:
yeah, why would the undo stack become the center of the universe here? This should be a core function, ideally callable from C code.